static/js/lib/ckeditor/attributeChecks.ts

```diff
@@ -0,0 +1,48 @@
+import { SchemaContext } from "@ckeditor/ckeditor5-engine/src/model/schema"
+
+/**
+ * Adds a check that can disallow the given attribute in the current context.
+ * This check is in addition to any default checks performed by the existing
+ * schema definition.
+ *
+ * See [CKEditor docs](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_model_schema-Schema.html#function-addAttributeCheck)
+ * for more details.
+ */
+type AttributeCheckFunction = (
+  context: SchemaContext,
+  attributeName: string
+) => boolean | undefined
+
+const someItemHasAttribute = (
+  context: SchemaContext,
+  attributeName: string
+): boolean => {
+  for (const item of context) {
+    for (const attribute of item.getAttributeKeys()) {
+      if (attribute === attributeName) {
+        return true
+      }
+    }
+  }
```

Comment:
Above, `context` and `item.getAttributeKeys()` are non-array iterables, hence the for loops as opposed to `Array.prototype.some`.